api/src/main/java/io/grpc/InternalConfigSelector.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import javax.annotation.Nullable;
+
+// The class can not be located in io.grpc.internal since it is used as a cross-module API.
+// Otherwise, shading would break it.
+/**
+ * Per method config selector that the channel or load balancers will use to choose the appropriate
+ * Route and MethodConfig for an RPC.
```

Comment:
ConfigSelector is not an xDS-specific design, its description should be more general. Same for the other places. (You are talking about xDS's concrete implementation `XdsConfigSelector`)

Reply:
Rephrased.